airflow/configuration.py

```diff
@@ -403,6 +403,52 @@ def inversed_deprecated_sections(self):
     upgraded_values: dict[tuple[str, str], str]
     """Mapping of (section,option) to the old value that was upgraded"""
 
+    legacy_incompatible_defaults: dict[tuple[str, str], tuple[Pattern, str, str]] = {
+        ("logging", "log_filename_template"): (
+            re.compile(
+                r"^(?:" + re.escape("{{ ti.dag_id }}/{{ ti.task_id }}/{{ ts }}/{{ try_number }}.log") + r"|"
+                r"dag_id=\{\s*ti\.dag_id\s*\}/run_id=\{\s*ti\.run_id\s*\}/task_id=\{\s*ti\.task_id\s*\}/"
+                r"\{%%\s*if\s+ti\.map_index\s*>=\s*0\s*%%\}map_index=\{\s*ti\.map_index\s*\}/\{%%\s*endif\s*%%\}"
+                r"attempt=\{\s*try_number\s*\}\.log" + r")$"
+            ),
+            "3.0",
+            "The default value for 'log_filename_template' from Airflow 2 may break task logs in the new UI.",
+        ),
+        ("core", "dag_ignore_file_syntax"): (
+            re.compile(r"^regexp$"),
+            "3.0",
+            "The default value changed from 'regexp' in Airflow 2.x to 'glob' in Airflow 3.0.",
+        ),
```

Comment:
```suggestion
```

Looking at this, automatically changing folks away from the default may break things, not using an old default. e.g. if their airflowignore was regex, but we automatically swap to glob, things won't be happy.

We should leave this one out I believe.